Stable build of 1.66.0 fails due to 'missing miri'

On Solaris 11 x86_64, from scratch `stable` builds of 1.66.0 from git fails with:
```
...
   Compiling opener v0.5.0
   Compiling toml v0.5.9
    Finished dev [unoptimized] target(s) in 1m 02s
Build completed successfully in 0:01:10
Building rustbuild
    Finished dev [unoptimized] target(s) in 2.25s
thread 'main' panicked at 'missing miri', install.rs:205:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Build completed unsuccessfully in 0:00:03
make: *** [Makefile:54: install] Error 1
```

A `nightly` build of the same tag completes successfully, as do `stable` and `nightly` builds of 1.65.0.

--- a/src/bootstrap/install.rs
+++ b/src/bootstrap/install.rs
@@ -200,10 +200,14 @@ install!((self, builder, _config),
         install_sh(builder, "clippy", self.compiler.stage, Some(self.target), &tarball);
     };
     Miri, alias = "miri", Self::should_build(_config), only_hosts: true, {
-        let tarball = builder
-            .ensure(dist::Miri { compiler: self.compiler, target: self.target })
-            .expect("missing miri");
-        install_sh(builder, "miri", self.compiler.stage, Some(self.target), &tarball);
+        if let Some(tarball) = builder.ensure(dist::Miri { compiler: self.compiler, target: self.target }) {
+            install_sh(builder, "miri", self.compiler.stage, Some(self.target), &tarball);
+        } else {
+            // Miri is only available on nightly
+            builder.info(
+                &format!("skipping Install miri stage{} ({})", self.compiler.stage, self.target),
+            );
+        }
     };
     LlvmTools, alias = "llvm-tools", Self::should_build(_config), only_hosts: true, {
         let tarball = builder

